src/config.ts

```diff
@@ -1,9 +1,15 @@
+import {Config, ConfigFn} from '../types/config'
+import {Callback} from '../types/utils'
 import {prettyDOM} from './pretty-dom'
 
+interface InternalConfig {
+  _disableExpensiveErrorDiagnostics: boolean
+}
+
 // It would be cleaner for this to live inside './queries', but
 // other parts of the code assume that all exports from
 // './queries' are query functions.
-let config = {
+let config: Config & InternalConfig = {
```

Comment:
Can you extend `Config` instead with `interface InternalConfig extends Config`? 

(Intersection) types are slightly slower than interfaces: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections